frontends/pytorch/csrc/c10_dispatch/module_builder.h

```diff
@@ -0,0 +1,50 @@
+//===- module_builder.h -----------------------------------------*- C++ -*-===//
+//
+// This file is licensed under a pytorch-style license
+// See frontends/pytorch/LICENSE for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef NPCOMP_FRONTENDS_PYTORCH_CSRC_C10_DISPATCH_MODULE_BUILDER_H
+#define NPCOMP_FRONTENDS_PYTORCH_CSRC_C10_DISPATCH_MODULE_BUILDER_H
+
+#include <pybind11/pybind11.h>
+
+#include "acap_dispatch.h"
+
+#include "mlir-c/IR.h"
+#include "llvm/ADT/SmallVector.h"
+
+namespace torch_mlir {
+
+/// Main entry-point for constructing an MLIR module from some combination
+/// of PyTorch programs/execution.
+class ModuleBuilder {
+public:
+  ModuleBuilder();
+  ~ModuleBuilder();
+
+  /// Creates Python bindings for the class.
+  static void bind(pybind11::module &m);
+
+  // TODO: Remove this once the MLIR Python objects are exposed directly.
+  pybind11::str getAsm();
+
+  // Starts a device-capture based function.
+  // TODO: Add inputs.
+  std::shared_ptr<AcapController> captureFunction(std::string &name);
```

Comment:
Nit: I found this name kind of confusing. It makes sense at the python level as a context manager. No action required, but if you have a better idiom here to make the C++ side of things clearer that would be nice.